### DISCUSSION TITLE
ui.json & Git merge

### DISCUSSION THREAD

**Discussion by JE-VS** [writer/writer-framework#377] [Q&A]
Hi all,

I have a question regarding the `ui.json` file during a Git merge. 

Two branches were created to work on two different pages (obviously, with their own GUI components, backend functions, ...). So far, so good. 

However, when performing a merge of the two branches, Git got totally confused with the `ui.json` file, due to the highly similar structure of each component description. 

I solved it by doing some JSON rebuilding by hand, but it is clearly not the way to go. 

I was then wondering if there was a good practice or recommendations when working with different branches ? Or simply something that I missed ? How do you do to handle or avoid this situation ? 

Thanks a lot! Best,

**Comment by FabienArcellier** [comment]
Hi @JE-VS,

Thanks for the feedback, we are aware of the limitations of the ui.json file. We plan to work on that for next release.

We think of implementing several changes :
* using one line in the json file per component (less lisible but conflict are easier to manage)
* using another format as [json line](https://jsonlines.org/) to reduce boilerplate
* split ui.json in several file (one per page)

Unfortunately, I don't know easy strategy to handle merge conflict today. When it happen, I have to rebuild the file by hand. 

I will try to publish an online tool to merge 2 ui.json file in one. The task is easy, it's just is not easy to handle through git. I let you know.

**Reply by JE-VS** [reply]
Thanks @FabienArcellier 

Great to read there are some thoughts on this issue! 

Meanwhile, I was also thinking about writing a short Python script to help for future merges. 
But if you have a tool, ready to try it too. 

Best,

**Reply by FabienArcellier** [reply]
I haven't finally, sorry about that. I don't think I will have time to develop this in the next few weeks.

**Reply by JE-VS** [reply]
No worries! Thanks anyway ;)
